frontend/src/lib/components/progressBar/ProgressBar.svelte

```diff
@@ -0,0 +1,85 @@
+<script lang="ts">
+	import { tweened } from 'svelte/motion'
+	import { cubicOut } from 'svelte/easing'
+	import { isLoop, type Progress } from './model'
+	import ProgressBarGeneralPart from './ProgressBarGeneralPart.svelte'
+	import ProgressBarLoopPart from './ProgressBarLoopPart.svelte'
+
+	export let steps: Progress = []
+	export let startIndex = 0
+	const length = 100 / steps.length
+	const percent = tweened(+(length * startIndex).toFixed(0), {
+		duration: 400,
+		easing: cubicOut
+	})
+	let currentIndex = startIndex
+	let loopIndex = 0
+	let isDone = false
+
+	function back() {
+		isDone = false
+		if (isLoop(currStep)) {
+			if (loopIndex <= 0) loopIndex = 0
+			else return loopIndex--
+		}
+
+		series[currentIndex].isDone = false
+		loopIndex = 0
+		if (currentIndex <= 0) currentIndex = 0
+		else currentIndex--
+		series[currentIndex].isDone = false
+	}
+
+	function next() {
+		if (isLoop(currStep)) {
+			const max = series[currentIndex].kind.length - 1
+			if (loopIndex >= max) loopIndex = max
+			else return loopIndex++
```

Comment:
I suggest using the bracket notation to increase readability

Reply:
On the `if else` lines, right?